src/main/java/com/conveyal/gtfs/graphql/fetchers/JDBCFetcher.java

```diff
@@ -435,26 +505,57 @@ private static void validateNamespace(String namespace) {
     }
 
     /**
-     * Construct filter clause with '=' (single string) and add values to list of parameters.
+     * Construct filter clause with '=' (single string) and add values to list of preparedStatementParameters.
      * */
-    static String makeInClause(String filterField, String string, List<String> parameters) {
-        // Add string to list of parameters (to be later used to set parameters for prepared statement).
+    static String filterEquals(String filterField, String string, List<String> parameters) {
+        // Add string to list of preparedStatementParameters (to be later used to set preparedStatementParameters for prepared sqlStatement).
         parameters.add(string);
         return String.format("%s = ?", filterField);
     }
 
     /**
-     * Construct filter clause with '=' (single string) or 'in' (multiple strings) and add values to list of parameters.
+     * Construct filter clause with '=' (single string) or 'in' (multiple strings) and add values to list of preparedStatementParameters.
      * */
     static String makeInClause(String filterField, List<String> strings, List<String> parameters) {
         if (strings.size() == 1) {
-            return makeInClause(filterField, strings.get(0), parameters);
+            return filterEquals(filterField, strings.get(0), parameters);
         } else {
-            // Add strings to list of parameters (to be later used to set parameters for prepared statement).
+            // Add strings to list of preparedStatementParameters (to be later used to set preparedStatementParameters for prepared sqlStatement).
             parameters.addAll(strings);
             String questionMarks = String.join(", ", Collections.nCopies(strings.size(), "?"));
             return String.format("%s in (%s)", filterField, questionMarks);
         }
     }
 
+    /**
+     * Helper class for batching sql queries to the dataloader
+     */
+    public static class JdbcQuery {
+        public String namespace;
+        public List<String> preparedStatementParameters;
+        public String sqlStatement;
+
+
+        public JdbcQuery(String namespace, List<String> preparedStatementParameters, String sqlStatement) {
+            this.namespace = namespace;
+            this.preparedStatementParameters = preparedStatementParameters;
+            this.sqlStatement = sqlStatement;
+        }
+
+        @Override
+        public boolean equals(Object o) {
```

Comment:
This could probably use a comment describing how it is used by the DataLoader

Reply:
And maybe also expand a bit on what's meant by "batching" sql queries. The fact that equals and hashcode are overridden make me think this has something to do with deduplicating the requests, i.e. this class serves as a key in a cache lookup. Is that the case? If so it deserves some clarification in the Javadoc.